BinaryTree.java

```diff
@@ -15,99 +24,133 @@ public BinaryTree(BinaryTreeNode<T> root) {
     }
 
     // Check if the key exists in the tree
-    public Boolean isExist(T key) {
-        return isExist(root, key);
+    public BinaryTreeNode exists(T key) {
+        return exists(root, key);
     }
 
     // Recursively check if the key exists in the tree (receives root)
-    public Boolean isExist(BinaryTreeNode<T> root, T key) {
-
-        if (root == null){
-            return false;
+    public BinaryTreeNode exists(BinaryTreeNode<T> root, T key) {

```

Comment:
Why not make `exists` a method in BinaryTreeNode, so that it can operate on `this` instead of explicitly passed root?
And again, containsKey would be a better name.